src/modules/cbm/CbmTypes.h

```diff
@@ -3,33 +3,90 @@
 ///
 /// \file
 ///
-/// \brief CBM beam monitor types
+/// \brief CBM beam monitor type definitions and conversion functions
 ///
 //===----------------------------------------------------------------------===//
 
 #pragma once
 
+#include <concurrentqueue.h>
 #include <stdexcept>
 #include <string>
 #include <sys/types.h>
 
 namespace cbm {
 
-// Implemented according to beam monitor ICD
-// https://project.esss.dk/nextcloud/index.php/f/17959238
+/// \class CbmType
+///
+/// \brief Represents the CBM (Common Beam Monitor type).
+///
+/// This class provides an enumeration of CBM beam monitor types and conversion
+/// functions to convert between different representations of the CBM type.
+///
+/// The CBM types are expressed as an enumeration of CbmTypes and the necessary
+/// conversion algorithms are provided by this class.
+///
+/// Furthermore this class also provides a to_string() function to convert the
+/// CBM type to a string representation.
+///
 
+/// Example usage:
+/// \code
+/// CbmType cbmType("TTL");
+/// int typeInt = static_cast<int>(cbmType);
+/// std::string typeStr = cbmType.to_string();
+/// 
+/// CbmType cbmType2("GEM");
+/// if (cbmType == cbmType2) {
+///     std::cout << "Both CBM types are the same." << std::endl;
+/// } else {
+///     std::cout << "CBM types are different." << std::endl;
+/// }
+/// \endcode
+
+/// \see CbmTypes
+/**
+ * @file CbmTypes.h
+ * @brief Defines the CbmType class and its associated enumeration CbmTypes.
+ */
+
```

Comment:
Just noticed this: We generally use the C++ variant for Doxygen comments. I did a quick search and it seems like the /** is something you have introduced

Please for consistency use /// instead
see for example src/common/kafka/AR51Serializer.h

Reply:
Yes, sorry it's comming from my JAVA habbits. I will take care for the future.